go/vt/vttablet/tabletmanager/restore.go

```diff
@@ -83,7 +83,23 @@ func (agent *ActionAgent) restoreDataLocked(ctx context.Context, logger logutil.
 	// Record local metadata values based on the original type.
 	localMetadata := agent.getLocalMetadataValues(originalType)
 	tablet := agent.Tablet()
-	dir := fmt.Sprintf("%v/%v", tablet.Keyspace, tablet.Shard)
+
+	// Backups are stored in a directory structure that starts with
+	// <keyspace>/<shard>
+	keyspaceDir := tablet.Keyspace
+	keyspaceInfo, err := agent.TopoServer.GetKeyspace(ctx, tablet.Keyspace)
+	if err != nil {
+		return err
+	}
+	// For a SNAPSHOT keyspace, we have to look for backups of BaseKeyspace
+	if keyspaceInfo.BaseKeyspace != "" {
+		keyspaceDir = keyspaceInfo.BaseKeyspace
+		// If we belong to a SNAPSHOT keyspace, let us disable_active_reparents
+		// and disable replication_reporter
+		*mysqlctl.DisableActiveReparents = false
+		*enableReplicationReporter = false
```

Comment:
can we add a log line noting that this was done

Reply:
Done